src/DIRAC/Core/Utilities/Decorators.py

```diff
@@ -82,10 +75,12 @@ def decFunc(func, clsName=None):
 
         decFunc.warningEn = True
 
-        if func.__doc__ is None:
-            func.__doc__ = "\n\n**Deprecated**: " + reason
-        else:
-            func.__doc__ += "\n\n**Deprecated**: " + reason
+        # If the __init__ function is inherited from an object, its attributes are not writable
+        if inspect.isfunction(func):
```

Comment:
is the `__init__` function not a function? How does this check prevent what is written in the comment?

Reply:
For `object` `__init__`(and others attributes, see `object.__dict__`) a [slot wrapper](https://doc.sagemath.org/html/en/reference/cpython/sage/cpython/wrapperdescr.html):
```python
>>> object.__init__
<slot wrapper '__init__' of 'object' objects>
```

and attributes of this type are not writable:
```python
>>> object.__init__.__doc__ = 'deprecated'
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: attribute '__doc__' of 'wrapper_descriptor' objects is not writable
```

it means that classes that inherit `object` has the same issue
```python
>>> class MyDeprecatedClass:
...   pass
... 
>>> class MyDeprecatedClassOldStyle(object):
...   pass
...
>>> MyDeprecatedClass.__init__ == MyDeprecatedClassOldStyle.__init__
True
>>> MyDeprecatedClass.__init__.__doc__ = 'deprecated'
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: attribute '__doc__' of 'wrapper_descriptor' objects is not writable
```

once you define `__init__` in your class as **function** you able to write attributes:
```python
>>> inspect.isfunction(MyDeprecatedClass.__init__)
False
>>> class ClassWithInit:
...   def __init__(self):
...     pass
... 
>>> inspect.isfunction(ClassWithInit.__init__)
True
>>> inspect.ismethod(ClassWithInit.__init__)
False
>>> ClassWithInit.__init__.__doc__
>>> ClassWithInit.__init__.__doc__ = 'deprecated'
>>> ClassWithInit.__init__.__doc__
'deprecated'
```

Reply:
I will update the comment, it is too simplistic

Reply:
fixed with https://github.com/DIRACGrid/DIRAC/pull/5870#discussion_r809001440

Reply:
Thanks for the explanation!